### PR TITLE
Fixes devices vector alloc to fix seg fault, removes unused RAFT code in PLC, re-enables full CI testing

### LINKED ISSUE #3124: CI crashes on pytest teardown
CI crashes with a seg fault on pytest teardown.

This is blocking cugraph from merging PRs.

### PR DESCRIPTION
**Pull request by rlratzel**
closes #3124 

* Adds check to avoid allocating and copying zero-length device vectors.  This prevents the seg fault shown below.
* Removes the special case to ignore seg faults in CI scripts
* Adds a test to reproduce seg fault locally (see output below).

This PR addresses the problem shown below:
```
================================= test session starts =================================
platform linux -- Python 3.8.15, pytest-7.2.0, pluggy-1.0.0 -- /opt/conda/envs/test/bin/python3.8
cachedir: .pytest_cache
rapids_pytest_benchmark: 0.0.14
benchmark: 3.2.3 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /cugraph/python/pylibcugraph, configfile: pytest.ini
plugins: cov-4.0.0, rapids-pytest-benchmark-0.0.14, benchmark-3.2.3
collected 9 items / 8 deselected / 1 selected

python/pylibcugraph/pylibcugraph/tests/test_graph_sg.py::test_SGGraph_create_from_cudf
get edgelist...edgelist =     src  dst  wgt
0    0    1  0.0
1    1    2  0.1
2    2    4  0.2
done
create Graph...done
created SGGraph plc_graph=<pylibcugraph.graphs.SGGraph object at 0x7fb7e35f30f0>
PASSED

=========================== 1 passed, 8 deselected in 1.69s ===========================
Segmentation fault (core dumped)
```

@cjnolet found a work-around for us, so this should pass CI and can be merged after https://github.com/rapidsai/raft/pull/1224

### CODE REVIEW

**Review by ajschmidt8 — APPROVED** [review]
:fire: :fire: :fire: :fire: 

Glad you guys got this figured out!

Approving `ops-codeowner` file changes.

**Review by ChuckHastings — APPROVED** [review]

File: python/pylibcugraph/CMakeLists.txt
Comment (rlratzel) [inline python/pylibcugraph/CMakeLists.txt:67]: Is this okay to do?  I thought DLFW builds (or something) needed this.

**Comment by rlratzel** [comment]
I added the DO NOT MERGE label so this isn't accidentally merged before commenting on if removing RAFT static linking still needs to be done: https://github.com/rapidsai/cugraph/pull/3167#discussion_r1094002853

File: python/pylibcugraph/CMakeLists.txt
Comment (rlratzel) [reply python/pylibcugraph/CMakeLists.txt]: ok thanks. I just reverted it in https://github.com/rapidsai/cugraph/pull/3167/commits/e51e86830df64ee0b56ef00622966d9a7aafb31b

**Comment by rlratzel** [comment]
Added DO NOT MERGE back until I've finished verifying it's safe to remove the RAFT sources from PLC, in the off chance CI passes even if they're needed.

**Comment by ChuckHastings** [comment]
Cancelled CI run.  We need https://github.com/rapidsai/raft/pull/1224 to be pushed into nightlies so that the change can be picked up by CI.

**Pull request merged**
